src/scripts/gamemodes/CaptureTheFlag/FlagStation.gd

```diff
@@ -0,0 +1,44 @@
+extends Area2D
+
+
+onready var Flag = preload("res://src/components/gamemodes/CaptureTheFlag/Flag.tscn")
+export var team_id = 0
+export var has_flag = true
+
+
+func _ready():
+# warning-ignore:return_value_discarded
+	GameEvent.connect("CTF_return_flag", self, "_on_CTF_return_flag")
+# warning-ignore:return_value_discarded	
+	GameEvent.connect("CTF_capture_flag", self, "_on_CTF_capture_flag")
+	if has_flag:
+		spawn_flag()
+
+
+func setup(_team_id):
+	team_id = _team_id
+
+
+func spawn_flag():
+	var FlagInstance = Flag.instance()
+	FlagInstance.setup(team_id)
+	call_deferred("add_child", FlagInstance)
```

Comment:
Maybe just add_child() is good enough

Reply:
The editor tells me to use call_deferred